api/openapi.yaml

```diff
@@ -145,49 +156,438 @@ paths:
+              schema:
+                properties:
+                  message:
+                    enum:
+                      - The LDAP credentials for this organization were invalid.
+                    type: string
+                type: object
+          description: Unauthorized
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        default:
+          $ref: '#/components/responses/DefaultResponse'
+      summary: Create a request to connect a user to an organization.
+      tags:
+        - stalker mobile app
+  '/user/{userId}/organizations/connections':
+    parameters:
+      - $ref: '#/components/parameters/userId'
+    get:
+      description: Get the list of organizations that a user has connected to.
+      operationId: get-user-userId-organizations-connections
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                properties:
+                  connectedOrganizations:
+                    items:
+                      $ref: ./model/GenericId.v1.yaml
+                    type: array
+                required:
+                  - connectedOrganizations
+                type: object
+          description: Ok. The response contains the list of the ids of the organizations the user is connected to.
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        default:
+          $ref: '#/components/responses/DefaultResponse'
+      summary: Read the list of all organizations a user is connected to.
+      tags:
+        - stalker mobile app
+  '/user/{userId}/places/history':
+    get:
+      description: "Get the history of a user's visited places."
+      operationId: get-user-userId-history
+      responses:
+        '204':
+          description: Not implemented yet.
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
         '501':
-          $ref: '#/components/responses/NotImplemented'
+          description: Unfinished endpoint.
         default:
           $ref: '#/components/responses/DefaultResponse'
-      summary: "Update a place's data."
+      summary: "Read a user's visited places history."
       tags:
-        - stalker web app
-  '/organization/{organizationId}/places':
+        - stalker mobile app
     parameters:
-      - $ref: '#/components/parameters/organizationId'
-    post:
-      description: Add a new place to the given organization. Operation reserved only to the managers and up.
-      operationId: postPlaceByOrganizationId
-      parameters:
-        - $ref: '#/components/parameters/organizationId'
-      requestBody:
-        content:
-          application/json:
-            schema:
-              $ref: ./model/PlaceData.v1.yaml
-        required: true
+      - $ref: '#/components/parameters/userId'
+  '/user/{userId}/time/inside':
```

Comment:
non servirebbe un path più specializzato? come identifichiamo a quale organizzazione e place ci riferiamo?
se facessimo così dovrebbero esserci tutti i dati di tutte le organizzazioni e tutti i places no?

Reply:
Hai ragione. io cambierei il response body in modo che dia la lista dei place in cui uno user è inside, e da quanto tempo è dentro a ciascun place

Reply:
ma dici in questo momento? perché è difficile che sia dentro in più places contemporaneamente

Reply:
forse c'è poca chiarezza su cosa è richiesto dal capitolato. 

![image](https://user-images.githubusercontent.com/30255227/80585161-2354e780-8a13-11ea-95dd-c172a9255892.png)

Io l'avevo interpretato come

"voglio sapere che sono al lavoro da 3 ore e mezza" ovvero dentro a un place di un'organizzazione con timeInside 3h30m.

e NON "voglio sapere che da quando mi sono collegato all'organizzazione sono rimasto dentro ai posti di quell'organizzazione un totale di 124 ore". 

chiaramente entrambe le soluzioni sono fattibili, ma io andrei con la prima (che eventualmente è possibile ampliare in "time spent inside since YYYYMMDD-HHMM" aggiungendo un request body e che darebbe come risposta "hai passato 70 ore dentro all'organizzazione dal 12 gennaio ad oggi"

Reply:
no infatti, ma proprio per questo non so se abbia senso ritornare una lista di places e da quanto tempo è dentro

Reply:
> perché è difficile che sia dentro in più places contemporaneamente

per rispondere a questo: lascerei un array perché se ci sono due organizzazioni con place sovrapposti (ad es il caso di un coworking tra startup) e sei collegato a entrambe, non abbiamo modo di capire in quale delle due sei quindi ti segniamo inside a entrambe

Reply:
visto che non credo sia normale che si trovi in più places contemporaneamente

Reply:
ok va bene